packages/block-editor/src/components/link-control/test/link-value-transforms.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Internal dependencies
+ */
+import {
+	buildLinkValueFromData,
+	buildDataFromLinkValue,
+} from '../link-value-transforms';
+
+/**
+ * Maps the standard LinkControl values to a given data object.
+ * Complex mappings may supply an object with a `getter` and `setter` function
+ * which represent how to get the link value for the given property and how to
+ * set it back on the data object.
+ */
+const mapping = {
+	url: 'href',
+	type: 'postType',
+	id: 'id',
+	opensInNewTab: {
+		dataKey: 'linkTarget',
+		toLink: ( value ) => value === '_blank',
+		toData: ( value ) => ( value ? '_blank' : undefined ),
+	},
+	noFollow: {
+		dataKey: 'linkRel',
+		toLink: ( value ) => value.includes( 'nofollow' ),
+		toData: ( value, currentVal ) => {
+			// if the value is truthy and the current value is set
+			// then append otherwise just add the value
+			if ( value && currentVal ) {
+				return `${ currentVal } nofollow`;
+			} else if ( value ) {
+				return 'nofollow';
+			}
```

Comment:
This type of operation feels like it could be fairly common. Should we look to provide some kind of utility to make this simpler and avoid boilerplate?